src/net/rpc.rs

```diff
@@ -559,10 +633,11 @@ impl ConversationHttp {
         sortdb: &SortitionDB,
         chainstate: &mut StacksChainState,
         tip: &StacksBlockId,
-        options: &ConnectionOptions,
+        burnchain: &Burnchain,
     ) -> Result<(), net_error> {
         let response_metadata = HttpResponseMetadata::from(req);
-        match RPCPoxInfoData::from_db(sortdb, chainstate, tip, options) {
+
+        match RPCPoxInfoData::from_db(sortdb, chainstate, tip, burnchain) {
```

Comment:
If we're going to be updating this method anyway, can you take the opportunity to handle the `Err(net_error::NotFoundError)` variant as an HTTP 404, instead of an HTTP 500?  Right now, we do the latter in case of any error, even if the only problem is that the `tip=` query parameter refers to an unknown block.

Reply:
Updated in 9e28b0a3851de7495c240c53595c65c477497033

Reply:
thanks!